src/main/java/com/hazelcast/remotecontroller/HzDockerCluster.java

```diff
@@ -0,0 +1,306 @@
+        try {
+            for (Map.Entry<String, List<String>> entry : this.blockRulesOfContainers.entrySet()) {
+                String containerId = entry.getKey();
+                List<String> blockedIps = entry.getValue();
+                unblockInputFromContainers(containerId, blockedIps);
+            }
+            return true;
+        } catch (Exception e) {
+            LOG.error("Could not merge brains", e);
+            return false;
+        }
+    }
+
+    public void shutdown() throws IOException {
+        Iterator<GenericContainer> iterator = this.containers.values().iterator();
+        while (iterator.hasNext()) {
+            GenericContainer container = iterator.next();
+            String str = container.getLogs();
+            BufferedWriter writer = new BufferedWriter(new FileWriter(container.getContainerId() + ".logs", true));
+            writer.append(str);
+            writer.close();
+            container.stop();
+        }
+        this.containers.clear();
+    }
+
+    private boolean checkMembersAreInThisCluster(List<DockerMember> members) {
+        for (DockerMember member : members) {
+            if (!this.containers.containsKey(member.getContainerId())) {
+                LOG.warn("Member " + member + " is not in this docker cluster");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void blockInputFromContainers(GenericContainer containerToBeAffected, List<DockerMember> dockerMembers) throws IOException, InterruptedException {
+        for (DockerMember member : dockerMembers) {
+            GenericContainer container = this.containers.get(member.getContainerId());
+            String containerIpAddress = getContainerIpAddress(container);
+            execAsRootAndThrowOnError(containerToBeAffected, "iptables", "-A", "INPUT", "-s", containerIpAddress, "-j", "DROP");
+            addToBlockRule(containerToBeAffected.getContainerId(), containerIpAddress);
+        }
+    }
+
+    private static String getContainerIpAddress(GenericContainer container) {
+        Collection<ContainerNetwork> networks = container.getContainerInfo().getNetworkSettings().getNetworks().values();
+        if (networks.isEmpty()) {
+            throw new RuntimeException("Container " + container.getContainerId() + " is not connected to any network."
+            + " In order to split brain the container, it has to be connected to a network.");
+        }
+        ContainerNetwork network = networks.iterator().next();
+        return network.getIpAddress();
+    }
+
+    private void addToBlockRule(String containerId, String blockedIp) {
+        if (blockRulesOfContainers.containsKey(containerId)) {
+            blockRulesOfContainers.get(containerId).add(blockedIp);
+        } else {
+            List<String> blockedIps = new ArrayList<>();
+            blockedIps.add(blockedIp);
+            blockRulesOfContainers.put(containerId, blockedIps);
+        }
```

Comment:
can be simplified to 
```suggestion
        List<String> blockedIps = blockRulesOfContainers.get(containerId);
        if (blockedIps == null) {
            blockedIps = new ArrayList<>();
            blockRulesOfContainers.put(containerId, blockedIps);
        }
        blockedIps.add(blockedIp);
```

this of course wouldn't work if the map needs to be CHM

Reply:
Why would it work? I did not get that part.